task_2/join.go

```diff
@@ -1,71 +0,0 @@
-package task_2
-
-import (
-	"context"
-	"fmt"
-	"math/rand"
-	"sync"
-	"time"
-)
-
-func Start() {
-
-	var (
-		numOfChannels = rand.Intn(12) + 1
-		channels      []chan interface{}
-	)
-
-	fmt.Println(fmt.Sprintf("creating %d channels..", numOfChannels))
-	for i := 0; i < numOfChannels; i++ {
-		channels = append(channels, make(chan interface{}))
```

Comment:
In this case, it is better to use a little buffer to prevent possible lock later.